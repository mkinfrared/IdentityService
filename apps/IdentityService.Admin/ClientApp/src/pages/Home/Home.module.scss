.Home {
  .token {
    overflow-wrap: anywhere;
    text-align: justify;
  }
}
