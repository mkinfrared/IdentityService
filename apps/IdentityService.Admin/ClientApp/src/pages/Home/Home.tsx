import { memo } from "react";

import useUserSession from "hooks/useUserSession";
import userManager from "utils/userManager";

import css from "./Home.module.scss";

const Home = () => {
  const user = useUserSession();

  const logIn = () => {
    userManager.signinRedirect();
  };

  const logOut = () => {
    userManager.signoutRedirect();
  };

  const logToken = async () => {
    const userData = await userManager.getUser();

    // eslint-disable-next-line no-console
    console.log("token", userData);
  };

  return (
    <div className={css.Home} data-testid="Home">
      <code>{user ? user.access_token : "logged out"}</code>
      <div>
        <button type="button" onClick={logIn}>
          Login
        </button>
        <button type="button" onClick={logToken}>
          Log User Data
        </button>
        <button type="button" onClick={logOut}>
          Logout
        </button>
      </div>
    </div>
  );
};

export { Home };

export default memo(Home);
