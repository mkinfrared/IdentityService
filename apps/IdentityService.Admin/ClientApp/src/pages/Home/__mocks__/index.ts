export { default } from "./Home.mock";
