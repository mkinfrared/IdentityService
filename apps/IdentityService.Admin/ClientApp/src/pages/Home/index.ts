export { default } from "./Home";

export * from "./Home.type";
