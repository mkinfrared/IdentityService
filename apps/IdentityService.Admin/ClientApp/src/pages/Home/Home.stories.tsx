import { Meta, Story } from "@storybook/react";

import { Home } from "./Home";
import { HomeProps } from "./Home.type";
import css from "./Story.module.scss";

export default {
  title: "PAGES/Home",
  component: Home,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<HomeProps> = () => (
  <div className={css.Story}>
    <Home />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
