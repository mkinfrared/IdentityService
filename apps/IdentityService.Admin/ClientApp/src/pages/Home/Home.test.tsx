import { act, render } from "@testing-library/react";

import { Home } from "./Home";

jest.mock("shared/lib/hooks", () => ({ useUserSession: jest.fn() }));

describe("<Home />", () => {
  const Component = <Home />;

  it("should be defined", () => {
    expect(Home).toBeDefined();
  });

  it("should match the snapshot", async () => {
    const { container } = render(Component);

    await act(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it("should contain a data test id", async () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Home");

    await act(() => {
      expect(element).toBeDefined();
    });
  });
});
