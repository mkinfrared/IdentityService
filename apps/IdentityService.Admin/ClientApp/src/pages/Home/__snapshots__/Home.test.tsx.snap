// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match the snapshot 1`] = `
<div>
  <div
    class="Home"
    data-testid="Home"
  >
    <p
      class="token"
    >
      logged out
    </p>
    <div>
      <button
        type="button"
      >
        Log User Data
      </button>
    </div>
  </div>
</div>
`;
